### PR TITLE
resource/alicloud_sae_application: Fixed the update bug in field command_args, custom_host_alias, oss_mount_descs, config_map_mount_desc, liveness, readiness, post_start, pre_stop, tomcat_config, update_strategy

### PR DESCRIPTION
**Pull request by MrWolong**

### CODE REVIEW

File: alicloud/resource_alicloud_sae_application.go [hunk 1]
```diff
@@ -2434,434 +2435,424 @@ func resourceAliCloudSaeApplicationUpdate(d *schema.ResourceData, meta interface
 		deployApplicationReq["ImagePullSecrets"] = StringPointer(v.(string))
 	}
 
-	if !d.IsNewResource() && (d.HasChange("command_args_v2") || d.HasChange("command_args")) {
```
Comment (ChenHanZhang) [inline alicloud/resource_alicloud_sae_application.go:2437]: 这里把 commad_args 去掉是什么原因？另外针对bug修复需要补充一个bugfix用例

**Review by ChenHanZhang — COMMENTED** [review]
approved

File: alicloud/resource_alicloud_sae_application.go [hunk 2]
```diff
@@ -2434,434 +2435,424 @@ func resourceAliCloudSaeApplicationUpdate(d *schema.ResourceData, meta interface
...
+	if !d.IsNewResource() && d.HasChange("command_args_v2") {
```
Comment (MrWolong) [inline alicloud/resource_alicloud_sae_application.go:2438]: 1、创建：
Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

command_args = "[\"1d\"]"
2、更新：
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

command_args = "[\"2d\"]"
3、plan
No changes. Your infrastructure matches the configuration.

**Review by ChenHanZhang — COMMENTED** [review]
approved

File: alicloud/resource_alicloud_sae_application.go [hunk 3]
```diff
@@ -227,10 +227,11 @@ func resourceAliCloudSaeApplication() *schema.Resource {
 				Computed: true,
 			},
 			"command_args_v2": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Computed: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+				Type:          schema.TypeList,
```
Comment (MrWolong) [inline alicloud/resource_alicloud_sae_application.go:230]: ci已pass
2024-04-22 10:00:57.993 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicFatJar -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicFatJar
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicFatJar (3324.24s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	3324.336s

2024-04-22 10:59:09.538 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicImage -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicImage
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicImage (2801.08s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	2801.179s

2024-04-22 11:46:05.159 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicNewFatJar -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicNewFatJar
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicNewFatJar (7066.66s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	7066.756s

2024-04-22 13:44:05.444 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicNewImage -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicNewImage
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicNewImage (2165.95s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	2166.044s

2024-04-22 14:20:25.316 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicPhp -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicPhp
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicPhp (2584.54s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	2584.631s

2024-04-22 15:03:43.573 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicTags -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicTags
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicTags (184.27s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	184.361s

2024-04-22 15:07:01.267 terraform-provider-qa/helper/run_command.go:22: [DEBUG] running command go-test ...
$ bash -c go test ./alicloud -v -run=TestAccAliCloudSAEApplication_basicWar -timeout=1200m

=== RUN   TestAccAliCloudSAEApplication_basicWar
    provider_test.go:149: Skipping unsupported region eu-central-1. Supported regions: [cn-hangzhou cn-shanghai cn-beijing cn-zhangjiakou cn-shenzhen us-west-1]. Using cn-hangzhou as this test region
--- PASS: TestAccAliCloudSAEApplication_basicWar (3493.18s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	3493.270s

**Pull request merged**